sys/net/application_layer/nanocoap/fileserver.c

```diff
@@ -143,10 +158,16 @@ static size_t gcoap_fileserver_error_handler(coap_pkt_t *pdu, uint8_t *buf, size
             default:
                 code = COAP_CODE_INTERNAL_SERVER_ERROR;
         };
-        DEBUG("gcoap_fileserver: Rejecting error %d (%s) as %d.%02d\n", err, strerror(err),
+        DEBUG("nanocoap_fileserver: Rejecting error %d (%s) as %d.%02d\n", err, strerror(err),
               code >> 5, code & 0x1f);
     }
-    return gcoap_response(pdu, buf, len, code);
+
+    if (_resp_init(pdu, buf, len, code)) {
+        return -1;
+    }
+    coap_opt_finish(pdu, COAP_OPT_FINISH_NONE);
+
+    return 0;
```

Comment:
Is gcoap_response maybe just as due for a rename?

Reply:
`gcoap_response` includes some observe handling which we would lose if we move the function to nanocoap. I'm not against adding a simple nanocoap_response in another PR though, as long as the difference with `gcoap` is documented somewhere.